trunk/SUAVE/Analyses/Constraint_Analysis/Constraint_Analysis.py

```diff
@@ -0,0 +1,1386 @@
+        eps       = self.landing.approach_speed_factor
+        delta_ISA = self.landing.delta_ISA
+        eng_type  = self.engine.type
+
+        # Estimate maximum lift coefficient
+        if cl_max == 0:
+            cl_max = self.estimate_max_lift(self.geometry.high_lift_type_takeoff)
+
+        # Estimate the approach speed
+        if eng_type != 'turbofan' and eng_type != 'turbojet':
+            kappa = 0.7
+            Vapp  = np.sqrt(3.8217*Sg/kappa+49.488)
+        else:
+            kappa = 0.6
+            Vapp  = np.sqrt(2.6319*Sg/kappa+458.8)
+
+        # Determine atmospheric properties at the altitude
+        planet      = SUAVE.Analyses.Planets.Planet()
+        atmosphere  = SUAVE.Analyses.Atmospheric.US_Standard_1976()
+        atmosphere.features.planet = planet.features
+
+        atmo_values = atmosphere.compute_values(altitude,delta_ISA)
+        rho         = atmo_values.density[0,0]
+
+        W_S        = np.zeros(2)
+        
+        cl_app     = cl_max / eps**2
+        W_S[:]     = 0.5*rho*Vapp**2*cl_app
+        constraint = np.array([0,30])
+
+        # Convert the wing loading depinting on the user input
+        if self.plot_units == 'US':
+            W_S = W_S * 0.020885              # convert to lb/sq ft
+        else:
+            W_S = W_S/9.81                    # convert to kg/sq m
+
+        return W_S,constraint
+
+
+
+    def oswald_efficiency(self,cdmin,mach):
+        """Calculate an average Oswald efficiencies based on different methods
+
+        Assumptions:
+        None
+
+        Source:
+           M. Nita, D.Scholtz, 'Estimating the Oswald factor from basic aircraft geometrical parameters',
+            Deutscher Luft- und Raumfahrtkongress 20121DocumentID: 281424
+
+        Inputs:
+            self.geometry.taper                 [Unitless]
+                 aspect_ratio                   [Unitless]
+                 sweep_quarter_chord            [radians]
+                 aerodynamics.fuselage_factor   [Unitless]
+                              viscous_factor    [Unitless]
+
+        Outputs:
+            e          [Unitless]
+
+        Properties Used:
+
+        """  
+
+        # Unpack inputs
+        taper = self.geometry.taper
+        AR    = self.geometry.aspect_ratio
+        sweep = self.geometry.sweep_quarter_chord / Units.degrees
+        kf    = self.aerodynamics.fuselage_factor 
+        K     = self.aerodynamics.viscous_factor 
+
+        dtaper    = -0.357+0.45*np.exp(-0.0375*np.abs(sweep))
+        eff_taper = taper - dtaper
+        f_taper   = 0.0524*eff_taper**4-0.15*eff_taper**3+0.1659*eff_taper**2-0.0706*eff_taper+0.0119
+        u         = 1 / (1+f_taper*AR)
+        P         = K*cdmin
+        Q         = 1/(kf*u)
+
+        #if mach > 0.3:
+        #    keM = -0.001521*(mach/0.3-1)**10.82+1
+        #else:
+        keM = 1
+        
+        e = keM/(Q+P*np.pi*AR)
+
+
+        return e
+
+
+    def estimate_max_lift(self,highlift_type):
```

Comment:
I think this could be useful for many other calculations. Can you make this more general and move to methods?

Reply:
I see that SUAVE already has the maximum lift estimation function. Is it better to have this function as a stand-alone or to integrate it into the SUAVE default maximum lift function?

Reply:
If you're able to integrate it, that would be great. You can make a setting that will switch between types.